sonic_installer/bootloader/aboot.py

```diff
@@ -114,11 +125,54 @@ def get_binary_image_version(self, image_path):
     def verify_binary_image(self, image_path):
         try:
             subprocess.check_call(['/usr/bin/unzip', '-tq', image_path])
-            # TODO: secureboot check signature
+            return self._verify_secureboot_image(image_path)
         except subprocess.CalledProcessError:
             return False
+
+    def _verify_secureboot_image(self, image_path):
+        if isSecureboot():
+            current_image_path = self.get_current_image()
+            current_swi_path = self._swi_image_path(current_image_path).replace('flash:', HOST_PATH + '/')
+            cert = self.getCert(image_path)
+            current_cert = self.getCert(current_swi_path)
+            if not cert or not current_cert:
+                return False
+            # Verify the signing certificates are from the same issuer
+            return str(cert.get_issuer()) == str(current_cert.get_issuer())
```

Comment:
I guess there is use case of change issuer in long run. Could you comment out this part of code?

Reply:
@qiluo-msft , we only need to verify the image has been signed, not need to verify the issuer, right?